common/djangoapps/third_party_auth/samlproviderdata/views.py

```diff
@@ -0,0 +1,54 @@
+"""
+    Viewset for auth/saml/v0/samlproviderdata
+"""
+
+from edx_rbac.mixins import PermissionRequiredMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from enterprise.models import EnterpriseCustomerIdentityProvider
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from openedx.features.enterprise_support.utils import fetch_enterprise_customer_by_id
+
+from ..models import SAMLProviderConfig, SAMLProviderData
+from .serializers import SAMLProviderDataSerializer
+
+
+class SAMLProviderDataMixin(object):
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = SAMLProviderDataSerializer
+
+
+class SAMLProviderDataViewSet(PermissionRequiredMixin, SAMLProviderDataMixin, viewsets.ModelViewSet):
+    """
+    A View to handle SAMLProviderData CRUD.
+    Uses the edx-rbac mixin PermissionRequiredMixin to apply enterprise authorization
+
+    Usage:
+        [HttpVerb] /auth/saml/v0/providerdata/
+    """
+    permission_required = 'enterprise.can_access_admin_dashboard'
+
+    def get_queryset(self):
+        """
+        Find and return the matching providerid for the given enterprise uuid
```

Comment:
nit: update this string, it's also not returning a providerid, this returns SAMLProviderData base on some other info